Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -0,0 +1,515 @@
+    ///   - maxReplacements: A number specifying how many occurrences of `other`
+    ///   to replace. Default is `Int.max`.
+    public mutating func replace<S: Sequence, Replacement: Collection>(
+        _ other: S,
+        with replacement: Replacement,
+        maxReplacements: Int = .max
+    ) where S.Element == Element, Replacement.Element == Element
+}
+
+extension RangeReplaceableCollection where SubSequence == Substring {
+    /// Returns a new collection in which all occurrences of a sequence matching
+    /// the given regex are replaced by another collection.
+    /// - Parameters:
+    ///   - regex: A regex describing the sequence to replace.
+    ///   - replacement: The new elements to add to the collection.
+    ///   - subrange: The range in the collection in which to search for `regex`.
+    ///   - maxReplacements: A number specifying how many occurrences of the
+    ///   sequence matching `regex` to replace. Default is `Int.max`.
+    /// - Returns: A new collection in which all occurrences of subsequence
+    /// matching `regex` in `subrange` are replaced by `replacement`.
+    public func replacing<R: RegexProtocol, Replacement: Collection>(
+        _ regex: R,
+        with replacement: Replacement,
+        subrange: Range<Index>,
+        maxReplacements: Int = .max
+    ) -> Self where Replacement.Element == Element
+  
+    /// Returns a new collection in which all occurrences of a sequence matching
+    /// the given regex are replaced by another collection.
+    /// - Parameters:
+    ///   - regex: A regex describing the sequence to replace.
+    ///   - replacement: The new elements to add to the collection.
+    ///   - maxReplacements: A number specifying how many occurrences of the
+    ///   sequence matching `regex` to replace. Default is `Int.max`.
+    /// - Returns: A new collection in which all occurrences of subsequence
+    /// matching `regex` are replaced by `replacement`.
+    public func replacing<R: RegexProtocol, Replacement: Collection>(
+        _ regex: R,
+        with replacement: Replacement,
+        maxReplacements: Int = .max
+    ) -> Self where Replacement.Element == Element
+  
+    /// Replaces all occurrences of the sequence matching the given regex with
+    /// a given collection.
+    /// - Parameters:
+    ///   - regex: A regex describing the sequence to replace.
+    ///   - replacement: The new elements to add to the collection.
+    ///   - maxReplacements: A number specifying how many occurrences of the
+    ///   sequence matching `regex` to replace. Default is `Int.max`.
+    public mutating func replace<R: RegexProtocol, Replacement: Collection>(
+        _ regex: R,
+        with replacement: Replacement,
+        maxReplacements: Int = .max
+    ) where Replacement.Element == Element
+  
+    public func replacing<R: RegexProtocol, Replacement: Collection>(
+        _ regex: R,
+        with replacement: (RegexMatch<R.Match>) throws -> Replacement,
```

Comment:
@timvermeulen @milseman I need help for documentation here. I'm not sure what this closure does.

Reply:
The closure receives the full match info, including captures, and returns a replacement string.